### PR TITLE
Retrocatively generate warning for outdated docs

### PR DESCRIPTION
**Pull request by pfitzseb**
Companion to https://github.com/JuliaDocs/Documenter.jl/pull/1577.

This uses the default light theme and a bunch of inline elements to make sure it works with whatever theme the old docs might've been using (e.g. mkdocs).

### CODE REVIEW

**Comment by mortenpi** [comment]
This is awesome!

1. I did notice that somehow the `x` is not visible with the old Documenter HTML theme:

   Screenshot from 2021-05-27 20-17-13

   Probably some weird interaction with the CSS there. For the new theme and MkDocs it seems to work perfectly.

2. It skips the latest version? I guess the idea is that you'd fix that on `master`? But I think  you might end up missing a fix for a version (e.g. `0.5.1` is the current latest, where it doesn't get applied, but the next tag is `0.6.0`, which will be the first one that has it from Documenter). Could we just detect if the tag is there or not already?

File: src/OutdatedWarning.jl [hunk 1]
```diff
@@ -0,0 +1,175 @@
+module OutdatedWarning
```
Comment (mortenpi) [inline src/OutdatedWarning.jl:1]: We could perhaps export the module, and then the user can just run this with:

```
julia> using DocumenterTools
julia> OutdatedWarning.generate("gh-pages-directory")
```

**Comment by pfitzseb** [comment]
I can't repro the styling issue:
image

The other comments should be addressed, but Windows CI chokes on the symlink. Will figure out what's going on with that...

**Review by mortenpi — APPROVED** [review]
LGTM! The styling issue seems to be gone now too (even when building with the old commit), so I am not sure what was causing that.

Edit: Oh, I _can_ repro the styling issue with the d00cf105cae4d9608bdfd630a9dd293fb5f2edde commit. So I think you already managed to fix the problem between me checking out this branch and posting the feedback :stuck_out_tongue:

**Pull request merged**